File: 240417/특정 수와 근접한 합/sum-close-to-particular-number.java
import java.util.*;
public class Main {
    public static int[] arr = new int[101];
    public static int n;
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        n = sc.nextInt();
        int S = sc.nextInt();

        for (int i = 0; i < n; i++){
            arr[i] = sc.nextInt();
        }

        int sum = 0;
        for (int e : arr) sum += e;
        int temp;
        int ans = Integer.MAX_VALUE;
        for (int i = 0; i < n; i++){
            for (int j = i+1; j < n; j++){
                temp = sum - arr[i] - arr[j];
                ans = Math.min(ans, Math.abs(S- temp));
            }
        }
        System.out.println(ans);
    }
}

File: 240422/특정 수와 근접한 합/sum-close-to-particular-number.py
N, S = map(int, input().split())
arr = list(map(int, input().split()))

total = sum(arr)

ans = int(1e9)
for i in range(N):
    for j in range(i+1, N):
        ans = min(ans, abs(S - (total - arr[i] - arr[j])))
print(ans)